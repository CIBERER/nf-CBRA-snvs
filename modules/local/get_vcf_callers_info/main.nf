process GET_VCF_CALLERS_INFO {
  
    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
      'https://depot.galaxyproject.org/singularity/ubuntu:20.04' :
      'nf-core/ubuntu:20.04' }"

    input:
    tuple val(meta), path(gt_file)

    output:
    tuple val(meta), path("${prefix}_SF.txt"), emit: sf_file
    tuple val(meta), path("*.txt"), emit: column_files
    path  "versions.yml"          , emit: versions


    script:

    prefix = task.ext.prefix ?: "${meta.id}"

    """
    # Extract the header line
    HEADER=\$(head -n 1 ${gt_file})

    # Count the number of columns
    COLS=\$(echo "\$HEADER" | awk '{print NF}')

    # Initialize an array to store file names
    FILES=()

    # Create files for each column based on the headers
    for ((i=1; i<=\$COLS; i++)); do
      # Get the header name for the current column
      COLUMN_HEADER=\$(echo "\$HEADER" | cut -f \$i)

      # Extract the column data (skip the header row)
      tail -n +2 ${gt_file} | cut -f \$i | awk -v header="\$COLUMN_HEADER" '{if (\$1 != "") print header; else print ""}' > "\${COLUMN_HEADER}.txt"

      # Add the file name to the array
      FILES+=("\${COLUMN_HEADER}.txt")
    done

    # Use paste to combine all the generated files and format the output with underscores
    paste -d "\t" "\${FILES[@]}" | perl -alne 'print join "_", @F' > ${prefix}_SF.txt

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        perl: \$(perl --version | head -n 2 | tail -n 1 | cut -d' ' -f4)
    END_VERSIONS
    """
}