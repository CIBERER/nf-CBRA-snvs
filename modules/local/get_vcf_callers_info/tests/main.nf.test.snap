{
    "get_vcf_callers_info - gt_file ": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "test_sample_SF.txt:md5,777c9906922e7d803c1f915388d43a9c"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test_sample"
                        },
                        [
                            "dragen.txt:md5,f210268825361009fb989c2c1d4c79b0",
                            "gatk.txt:md5,eb8969c52f9ad57f7dfea72ac14a88dd",
                            "test_sample_SF.txt:md5,777c9906922e7d803c1f915388d43a9c"
                        ]
                    ]
                ],
                "2": [
                    "versions.yml:md5,f28b81f2210329f01b7eed95730f449d"
                ],
                "column_files": [
                    [
                        {
                            "id": "test_sample"
                        },
                        [
                            "dragen.txt:md5,f210268825361009fb989c2c1d4c79b0",
                            "gatk.txt:md5,eb8969c52f9ad57f7dfea72ac14a88dd",
                            "test_sample_SF.txt:md5,777c9906922e7d803c1f915388d43a9c"
                        ]
                    ]
                ],
                "sf_file": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "test_sample_SF.txt:md5,777c9906922e7d803c1f915388d43a9c"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,f28b81f2210329f01b7eed95730f449d"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.04.4"
        },
        "timestamp": "2025-05-05T10:20:48.953825394"
    }
}